### PR TITLE
gherkin/javascript: Include tags from rules in scenario outlines

### PR DESCRIPTION
**Pull request by chaofan-nirvana**
See #2090

### CODE REVIEW

**Review by luke-hill — COMMENTED** [review]
Seems legit. Good spot. Let a JS dev review / approve though

File: gherkin/javascript/src/pickles/compile.ts [hunk 1]
```diff
@@ -79,19 +79,13 @@ function compileRule(
...
-        stepsContainer.scenario,
-        language,
-        pickles,
-        uri,
-        newId
-      )
     } else {
-      compileScenarioOutline(
-        featureTags,
+      const compileFunc = stepsContainer.scenario.examples.length === 0
```
Comment (mpkorstanje) [inline gherkin/javascript/src/pickles/compile.ts:83]: While not as short, could you use the same pattern for this as in Ruby? Having the same structure makes it easier to compare across language's.

**Review by mpkorstanje — COMMENTED** [review]
Small comment on style, please also update the changelog.

File: gherkin/javascript/src/pickles/compile.ts (hunk 1 above)
Comment (chaofan-nirvana) [reply gherkin/javascript/src/pickles/compile.ts:83]: Sure, but I am on my vacation without a laptop. I will do this asap！ Thanks！

Comment (chaofan-nirvana) [reply gherkin/javascript/src/pickles/compile.ts]: Sorry for the delay of such a long time. I just committed a patch.

**Comment by aslakhellesoy** [comment]
Hi @chaofan-nirvana,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by ciaranmcnulty** [comment]
Is it possible to add a new, or modify an existing, feature file from the testdata that would expose this difference?

**Comment by luke-hill** [comment]
Yes, we should. This one: https://github.com/cucumber/common/blob/main/gherkin/testdata/good/rule_with_tag.feature

**Comment by mpkorstanje** [comment]
I've got an MR for that in progress. It's not trivial, the code to generate the reference files is broken.

**Comment by mpkorstanje** [comment]
Available in Gherkin 24.1.

**Comment by chaofan-nirvana** [comment]
The cucumber and its community are so great! It is a good start for me. Thank you all!

**Pull request merged**